src/vm/methodtablebuilder.cpp

```diff
@@ -10235,6 +10236,12 @@ void MethodTableBuilder::CheckForSystemTypes()
             pMT->ClearFlag(MethodTable::enum_flag_Category_Mask);
             pMT->SetInternalCorElementType(ELEMENT_TYPE_VALUETYPE);
 #endif
+
+            // Fix the element type of the by-ref field
+            _ASSERTE(pMT->GetNumIntroducedInstanceFields() == 1);
+            PTR_FieldDesc byRefFieldDesc = pMT->GetFieldDescByIndex(0);
+            _ASSERTE(byRefFieldDesc->GetFieldType() == ELEMENT_TYPE_I);
+            byRefFieldDesc->SetFieldType(ELEMENT_TYPE_BYREF);
```

Comment:
I think it would be better to compensate for this on the JIT-EE interface just to make the JIT happy. It is very hard to make this self-consistent in the VM, e.g. the parts of the VM that access metadata directly will still see the IntPtr and not byref.